src/libmongoc/tests/test-mongoc-client-pool.c

```diff
@@ -361,15 +361,43 @@ test_client_pool_destroy_without_pushing (void)
    bson_destroy (cmd);
 }
 
+/* tests that creating and destroying session is ok */
+static void
+test_client_pool_create_unused_session (void)
+{
+   mongoc_client_t *client;
+   mongoc_client_pool_t *pool;
+   mongoc_uri_t *uri;
+   mongoc_client_session_t *session;
+
+   bson_error_t error;
+
+   capture_logs (true);
+
+   uri = mongoc_uri_new_with_error ("mongodb://127.0.0.1/", &error);
```

Comment:
I think this is the cause of the memory leak on evergreen [here](https://evergreen.mongodb.com/task_log_raw/mongo_c_driver_asan_ubuntu_test_asan_latest_server_auth_nosasl_openssl_cse_patch_43c5aaf21185156cc9e56ac25ac105c51f0a44be_5eebc80ec9ec44013775ff97_20_06_18_20_01_19/0?type=T#L7251). Because `mongoc_uri_new_with_error` calls `bson_malloc0`, which allocates and zeros out memory, and sets this uri to that address we have to free it. You should just be able to add `mongoc_uri_destroy (uri);`  below to fix it.